test/distributed/test_c10d_nccl.py

```diff
@@ -388,6 +389,24 @@ def test_nan_p2p(self):
         # reset env
         os.environ["TORCH_NCCL_NAN_CHECK"] = "0"
 
+    @requires_nccl()
+    @skip_if_lt_x_gpu(2)
+    def test_nan_check(self):
+        # Not expecting an error, NaN check should not make legit code fail
+        os.environ["TORCH_NCCL_NAN_CHECK"] = "1"
```

Comment:
Should we also set CUDA_LAUNCH_BLOCKING=1 in this test? as this test failed before only it is enabled

Reply:
I think I've seen test failing too when CUDA_LAUNCH_BLOCKING=0 (when the test has two all_reduce's).